### PR TITLE
Add optional title to rule groups

### PR DESCRIPTION
**Pull request by Hamish-Blinx**
Add an optional title to rule groups to be set by the onAddGroup function.

Use case:  We want to generate a search hitting a number of tables. Fields need to be grouped by the table and within a group only fields from the same table are to be selected. We are using the add group button to select the group and the onAddGroup function and onAddRule function to restrict the available fields. We want to add a title to the group so it is clear what sort it is for the user.

### CODE REVIEW

File: packages/react-querybuilder/src/types/propsUsingReact.ts
Comment (Hamish-Blinx) [inline packages/react-querybuilder/src/types/propsUsingReact.ts:1]: Sorry didn't intend to change this file, but can't seem to revert it now.

**Comment by jakeboone02** [comment]
Thanks for submitting this. It's kind of a niche feature, and since I'd like to keep the base rule group definition to a minimum I'm not going to merge it.

However, you've convinced me to finally add a feature I've been mulling over for some time: separate `controlElements` properties for rule group header and body components. This will make it quite a bit easier to implement requirements like the one you asked about it #751. You'd only need a small wrapper on the header element component instead of a reimplementation of the entire `RuleGroup` component.

One general git/GitHub-related comment on your PR: Avoid committing to the `main` branch of your own fork unless it will permanently diverge from the origin. This PR is actually a good example of why, since it won't be merged your `main` branch now differs from the origin `main` branch and you'll have to reset it to keep it in sync.

**Comment by Hamish-Blinx** [comment]
OK, many thanks for the quick response and help again. Thanks for the tip too. I'm a rookie as you can tell.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by vikassaini-it** [comment]
This should be a feature, not sure why this is left out, it looks really bad when a random name like "Rule Group at path 0"

**Comment by jakeboone02** [comment]
> This should be a feature, not sure why this is left out, it looks really bad when a random name like "Rule Group at path 0"

See [`accessibleDescriptionGenerator`](https://react-querybuilder.js.org/docs/components/querybuilder#accessibledescriptiongenerator).

**Pull request closed without merging**